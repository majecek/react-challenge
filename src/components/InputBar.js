import React, { Component } from 'react'

class InputBar extends Component {

  state = {
    inputTerm: ''
  }

  onInputChange = (event) => {
    console.log(event.target.value)
    this.setState({inputTerm: event.target.value})
  }

  onFormSubmit = (event) => {
    event.preventDefault()
    this.props.onInput(this.state.inputTerm.split(','))
  }

  render () {
    return (
      <div>
        <form onSubmit={this.onFormSubmit} className="input-group">
          <input
            placeholder="insert your array separated by commas"
            className="form-control"
            value={this.state.inputTerm}
            onChange={this.onInputChange}/>
          <button type="submit">Submit</button>
        </form>
      </div>
    )
  }
}

export default InputBar
