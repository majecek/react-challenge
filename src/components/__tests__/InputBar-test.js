import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import InputBar from './../InputBar'

it('InputBar component check against snapshot', () => {
  const onChange = jest.fn()
  const component = renderer.create(<InputBar onInput={onChange} />).toJSON()
  expect(component).toMatchSnapshot()
})

it('InputBar checks state properties', async () => {
  const onChange = jest.fn()
  const wrapper = mount(<InputBar onInput={onChange} />)
  // console.log(wrapper.state())

  expect(wrapper.state('inputTerm')).toEqual('')
  await wrapper.find('input').simulate('change')
  // console.log(wrapper.state())
  expect(wrapper.state('error')).toEqual('')
  // expect(onChange).toBeCalledWith(1);
})
