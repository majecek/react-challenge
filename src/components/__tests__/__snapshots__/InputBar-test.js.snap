// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBar component check against snapshot 1`] = `
<div>
  <form
    className="input-group"
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      onChange={[Function]}
      placeholder="insert your array separated by comma"
      value=""
    />
    <span
      className="input-group-btn"
    >
      <button
        className="btn btn-secondary"
        type="submit"
      >
        Submit
      </button>
    </span>
  </form>
  <div
    className="form-group has-danger"
  >
    <div
      className="form-control-feedback"
    >
      
    </div>
  </div>
</div>
`;
